BigQuery/tests/System/RegionalizationTest.php

```diff
@@ -0,0 +1,207 @@
+<?php
+/**
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\BigQuery\Tests\System;
+
+use Google\Cloud\Core\Exception\NotFoundException;
+use Google\Cloud\Core\Exception\ServiceException;
+
+/**
+ * @group bigquery
+ * @group bigquery-regionalization
+ */
+class RegionalizationTest extends BigQueryTestCase
+{
+    const REGION_ASIA = 'asia-northeast1';
+    const REGION_US = 'US';
+    const QUERY_TEMPLATE = 'SELECT 1 FROM `%s.%s`';
+
+    private static $datasetAsia;
+    private static $tableAsia;
+
+    public static function setUpBeforeClass()
+    {
+        parent::setUpBeforeClass();
+        self::$datasetAsia = self::createDataset(
+            self::$client,
+            uniqid(self::TESTING_PREFIX),
+            ['location' => self::REGION_ASIA]
+        );
+        self::$tableAsia = self::createTable(
+            self::$datasetAsia,
+            uniqid(self::TESTING_PREFIX)
+        );
+    }
+
+    /**
+     * @dataProvider locationDataProvider
+     */
+    public function testRunQuery($location)
+    {
+        $caught = false;
+        $query = self::$client->query(
+            sprintf(
+                self::QUERY_TEMPLATE,
+                self::$datasetAsia->id(),
+                self::$tableAsia->id()
+            )
+        )->location($location);
```

Comment:
Could you assert that there are results from the query, too? Maybe just that the number of rows in the results is > 0.

Reply:
Sure! The reason I omitted this (and other checks) is because they are already covered in other system tests, but I can see the value in having the checks here too.